source/common/dynamo/dynamo_filter.cc

```diff
@@ -208,4 +210,44 @@ void DynamoFilter::chargeFailureSpecificStats(const Buffer::Instance& data) {
   }
 }
 
+void DynamoFilter::chargeTablePartitionIdStats(const Buffer::Instance& data) {
+  // Only log partition stats for single table operations.
+  if (table_descriptor_.table_name.empty() || operation_.empty()) {
+    return;
+  }
+  std::string body = buildBody(encoder_callbacks_->encodingBuffer(), data);
+  if (!body.empty()) {
+    try {
+      std::vector<Dynamo::RequestParser::PartitionDescriptor> partitions =
+          Dynamo::RequestParser::parsePartitions(body);
+      for (const Dynamo::RequestParser::PartitionDescriptor& partition : partitions) {
+        std::string stats_string = DynamoFilter::buildPartitionStatString(
+            stat_prefix_, table_descriptor_.table_name, operation_, partition.partition_id_);
+        stats_.counter(stats_string).add(partition.capacity_);
+      }
+    } catch (const Json::Exception& jsonEx) {
+      // Body parsing failed. This should not happen, just put a stat for that.
+      stats_.counter(fmt::format("{}invalid_resp_body", stat_prefix_)).inc();
+    }
+  }
+}
+
+std::string DynamoFilter::buildPartitionStatString(const std::string& stat_prefix,
+                                                   const std::string& table_name,
+                                                   const std::string& operation,
+                                                   const std::string& partition_id) {
+  std::string stats_table_prefix = fmt::format("{}table.{}", stat_prefix, table_name);
+  // Use only the last 7 characters from the partition id.
+  std::string stats_partition_postfix =
+      fmt::format(".Capacity.{}.__partition_id={}", operation,
+                  partition_id.substr(partition_id.size() - 7, partition_id.size()));
+  // Calculate how many characters are available for the table prefix.
+  size_t remaining_size = MAX_NAME_SIZE - stats_partition_postfix.size() - 1;
```

Comment:
why do we have here -1, MAX_NAME_SIZE is the actual length we can have without, see RawStatData struct (it allocates MAX_NAME_SIZE+1 for \0).